docs/stateful_models.md

```diff
@@ -0,0 +1,317 @@
+   - 1 - indicates start of the sequence
+   - 2 - indicates end of the sequence
+
+**Note**: Model server also appends `sequence_id` to every response.
+
+**Both `sequence_id` and `sequence_control_input` must be provided as 1-dimensional arrays of size 1.**  
+_See examples for gRPC and HTTP below_.
+
+In order to successfully infer the sequence, perform these actions:
+1. **Send the first request in the sequence and signal sequence start.**
+
+   To start the sequence you need to add `sequence_control_input` with value of 1 to your request's inputs. You can also:
+      - add `sequence_id` with the value of your choice or
+      - add `sequence_id` with 0 or do not add `sequence_id` at all - in this case model server will provide unique id for the sequence and since it'll be appended to the outputs, you'll be able to read it and use with the next requests. 
+
+2. **Send remaining requests except the last one.**
+
+   To send requests in the middle of the sequence you need to add `sequence_id` of your sequence. In this case `sequence_id` is mandatory and not providing this input or setting it's value to 0 is not allowed.
+
+3. **Send the last request in the sequence and signal sequence end.**
+
+   To end the sequence you need to add `sequence_control_input` with the value of 2 to your request's inputs. You also need to add `sequence_id` of your sequence. In this case `sequence_id` is mandatory and not providing this input or setting it's value to 0 is not allowed.
+
+
+### Inference via gRPC <a name="stateful_grpc"></a>
+
+Inference on stateful models via gRPC is very similar to inference on stateless models (_see [gRPC API](model_server_grpc_api.md#predict-api) for reference_). The difference is that requests to stateful models must containt additional inputs with information necessary for proper sequence handling.
+
+`sequence_id` and `sequence_control_input` must be added to gRPC request inputs as [TensorProtos](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.proto).
+
+* For `sequence_id` model server expects one value in tensor proto [uint64_val](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.proto#L85) field.
+
+* For `sequence_control_input` model server expects one value in tensor proto [uint32_val](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.proto#L82) field.
+
+Both inputs must have `TensorShape` set to [1] and appropriate `DataType`:
+- `DT_UINT64` for `sequence_id`
+- `DT_UINT32` for `sequence_control_input`
+
+Example: (_using Python tensorflow and tensorflow-serving-api packages_):
+
+```
+...
+import grpc
+
+from tensorflow_serving.apis import prediction_service_pb2_grpc
+from tensorflow import make_tensor_proto, make_ndarray, expand_dims
+from tensorflow_serving.apis import predict_pb2
+
+...
+
+SEQUENCE_START = 1
+SEQUENCE_END = 2
+sequence_id = 10
```

Comment:
why is it in lower case?

Reply:
To indicate this value is not const. 
`SEQUENCE_START` and `SEQUENCE_END` are const and have hard coded values. 
`sequence_id` can be chosen by the user